nix/tests.nix

```diff
@@ -112,6 +113,58 @@ let
 
         ${withTmpDb postgresql} "$rootdir/test/memory-tests.sh"
       '';
+
+  coverage =
+    name: postgresql:
+    let
+      covCabalOptions="${devCabalOptions} --enable-coverage --enable-executable-dynamic";
```

Comment:
Without enabling `--enable-executable-dynamic` I had the problem that as soon as I built exe, lib and tests, every part was built twice. HPC flags seemed to change randomly. Whenever I ran the coverage job, it had to rebuild from scratch, even though I did not change any code.

The combination of `--enable-coverage --enable-execute-dynamic` solves that. The executable is now linked against a shared library. Maybe we should enable that for all dev builds? Maybe it is already - I am confused where those settings actually come from. `package.cabal`, `devCabalOptions`, `*.nix`, ...